README.md

```diff
@@ -178,405 +128,133 @@ After installing the library you will need to require it in your test file (see
 
-        const goToContactPageButton = element(by.className("contact-button"));
+[Example](docs/EXAMPLES.md#waitforelementpresence)
 
-        goToContactPageButton.click();
+### `waitForElementNotToBePresent`
 
-        protractorHelper.waitForUrlToBeEqualToExpectedUrl("https://example.com/contact", "URL is different from expected", 3000);
+That method is the opposite of the previous one, so, it waits for an element not to be present in the DOM.
+[Example](docs/EXAMPLES.md#waitforelementnottobepresent)
 
-        // ...
-    });
-});
-```
+### `waitForElementVisibility`
 
-### Example of usage of `waitForUrlNotToBeEqualToExpectedUrl`
+That method waits for an element to be visible in the page. Being displayed means not only that the element is present in the DOM, but also that is has a height and width that is greater than 0.
+[Example](docs/EXAMPLES.md#waitforelementvisibility)
 
-```js
-const protractorHelper = require("protractor-helper");
+### `waitForElementNotToBeVisible`
 
-describe("foo", () => {
-    it("bar", () => {
-        const homePageUrl = "https://example.com";
+That method is the opposite of the previous one, so, it waits for an element not to be visible in the page. By saying not being displayed means that the element may be in the DOM, but not visible.
+[Example](docs/EXAMPLES.md#waitforelementnottobevisible)
 
-        browser.get(homePageUrl);
+### `clickWhenClickable`
 
-        const goToContactPageButton = element(by.className("contact-button"));
+That method is used to click in an element only as soon as it is in a clickable state. This means that the element is visible and enabled for clicking.
+[Example](docs/EXAMPLES.md#clickwhenclickable)
 
-        goToContactPageButton.click();
+### `fillFieldWithTextWhenVisible`
 
-        protractorHelper.waitForUrlNotToBeEqualToExpectedUrl(homePageUrl, "URL is equal to " + homePageUrl, 3000);
+That method fills an input field with a text as soon as such field is visible.
+[Example](docs/EXAMPLES.md#fillfieldwithtextwhenvisible)
 
-        // ...
-    });
-});
-```
+### `fillInputFieldWithFileWhenPresent`
 
-### Example of usage of `waitForUrlToContainString`
+That method fills a file input field with a specified file as soon as the file input field is present in the DOM.
+[Example](docs/EXAMPLES.md#fillinputfieldwithfilewhenpresent)
 
-```js
-const protractorHelper = require("protractor-helper");
+### `clearFieldWhenVisible`
 
-describe("foo", () => {
-    it("bar", () => {
-        browser.get("https://example.com");
+That method clears a text input field as soon as such field is visible.
 
-        const goToContactPageButton = element(by.className("contact-button"));
+### `clearFieldWhenVisibleAndFillItWithText`
 
-        goToContactPageButton.click();
+That method clears a text input field as soon as such field is visible, and then it fills it with a text.
```

Comment:
Same as others.

Reply:
Commit made because this review: ce8ef21